packages/core/src/modules/dids/domain/DidDocument.ts

```diff
@@ -200,6 +200,14 @@ export class DidDocument {
     return recipientKeys
   }
 
+  public get agreementKeys(): Array<VerificationMethod> {
```

Comment:
This duplicates the separate method `getAgreementKeys ` below. If we add this for the keyAgreements, we should add it for all types.

Maybe we can do something like this. This is similar to `derefernceKey`, and is more general purpose. We can extract `allPurposes` as it's also used by `derefenceKey`

```ts
public dereferenceVerificationMethods(purposes?: DidPurpose[] | DidPurpose) {
  const allPurposes: DidPurpose[] = [
      'authentication',
      'keyAgreement',
      'assertionMethod',
      'capabilityInvocation',
      'capabilityDelegation',
    ]

    const verificationMethods: VerificationMethod[] = []
    for (const purpose of purposes ?? allPurposes) {
      for (const verificationMethod of this[purpose] ?? []) {
        if (typeof key === 'string') {
          verificationMethods.push(this.dereferenceVerificationMethod(verificationMethod))
        } else {
          verificationMethods.push(verificationMethod)
        }
      }
    }

return verificationMethods
}
```

Reply:
Removed duplicate method.

Reply:
My comment is still valid. We only have this method for keyAgreement keys now, not for the other purposes. I think we should make the methods on the did document generic (so it works with all purposes)

Reply:
Added suggested method. This method is almost the duplication of `dereferenceKey`...

Reply:
agreed. maybe we can reuse them at some point